src/component/segments/SegmentFormStepTwo/SegmentFormStepTwo.tsx

```diff
@@ -0,0 +1,135 @@
+import PermissionButton from 'component/common/PermissionButton/PermissionButton';
+import { SidebarModal } from 'component/common/SidebarModal/SidebarModal';
+import { CreateContext } from 'component/context/CreateContext/CreateContext';
+import { CREATE_CONTEXT_FIELD } from 'component/providers/AccessProvider/permissions';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+import { IConstraint } from 'interfaces/strategy';
+import { useHistory } from 'react-router-dom';
+import { useStyles } from 'component/segments/SegmentFormStepTwo/SegmentFormStepTwo.styles';
+import {
+    ConstraintAccordionList,
+    IConstraintAccordionListRef,
+} from 'component/common/ConstraintAccordion/ConstraintAccordionList/ConstraintAccordionList';
+import { SegmentFormStep } from '../SegmentForm/SegmentForm';
+import {
+    AutocompleteBox,
+    IAutocompleteBoxOption,
+} from 'component/common/AutocompleteBox/AutocompleteBox';
+
+interface ISegmentFormPartTwoProps {
+    constraints: IConstraint[];
+    setConstraints: React.Dispatch<React.SetStateAction<IConstraint[]>>;
+    setCurrentStep: React.Dispatch<React.SetStateAction<SegmentFormStep>>;
+}
+
+export const SegmentFormStepTwo: React.FC<ISegmentFormPartTwoProps> = ({
+    children,
+    constraints,
+    setConstraints,
+    setCurrentStep,
+}) => {
+    const constraintsAccordionListRef = useRef<IConstraintAccordionListRef>();
+    const history = useHistory();
+    const styles = useStyles();
+    const { context = [] } = useUnleashContext();
+    const [open, setOpen] = useState(false);
+
+    const autocompleteOptions = context.map(c => ({
+        value: c.name,
+        label: c.name,
+    }));
+
+    const onChange = ([option]: IAutocompleteBoxOption[]) => {
+        constraintsAccordionListRef.current?.addConstraint(option.value);
+    };
+
+    return (
+        <div className={styles.form}>
+            <div className={styles.container}>
+                <h3 className={styles.formHeader}>
+                    Select the context fields you want to include in the segment
+                </h3>
+                <div>
+                    <p className={styles.inputDescription}>
+                        Use a predefined context field
+                    </p>
+                    <AutocompleteBox
+                        label="Select a context"
+                        options={autocompleteOptions}
+                        onChange={onChange}
+                    />
+                </div>
+
+                <div className={styles.addContextContainer}>
+                    <p className={styles.inputDescription}>
+                        Or create and add a new custom context field
+                    </p>
+                    <SidebarModal
+                        label="Create new context"
+                        onClose={() => setOpen(false)}
+                        open={open}
+                    >
+                        <CreateContext
```

Comment:
Context is such a loaded word. I frequently mix it with react context, golang contexts and unleash context. To add to the confusion, there are subtle differences conceptually. Maybe we should rename our internal context references to `CreateUnleashContext` like we have done with the hook?